proto/attestation/reference_value.proto

```diff
@@ -94,21 +95,34 @@ message Regex {
   string value = 1;
 }
 
+message StringLiteral {
+  string value = 1;
```

Comment:
I guess this may become repeated in the future, right?

Reply:
If it will changing this field to repeated will be rather painful, but if we don't already have a use case for multiple values I'd rather leave it as is... I guess if this becomes necessary we can add another oneof option, e.g., MultiStringLiteral. WDYT?

Reply:
For parity with [BinaryReferenceValue](https://github.com/project-oak/oak/blob/863ee006a3154a7f1f6d82b45230c4b4f36f8c85/proto/attestation/reference_value.proto#L53), which has a Digests submessage which lists [multiple](https://github.com/project-oak/oak/blob/863ee006a3154a7f1f6d82b45230c4b4f36f8c85/proto/attestation/reference_value.proto#L134) valid digests (only of which has to match), I would actually lean towards using multiple string values here, as Tiziano suggests.

We're currently operating under the assumption that we can create one ReferenceValues proto to verify a set of possibly TEE configurations, and due to the multiple allowable digests per layer this works right now, but if the kernel command line were to vary across configurations that would break down with this current "single literal" approach.

Reply:
Hmmm, do we have a story about how to match a particular kernel cmd line to a particular binary to define an acceptable TEE configuration? Or would you expect to define a set of binaries + a set of command lines where any combination is acceptable?

Reply:
That's currently the assumption we're making, yeah, that we don't have to strictly verify specific configurations. E.g. that given:

- layer 1: A or B is allowed
- layer 2: C or D is allowed

means that all of the following configurations are allowable:
- A && C
- A && D
- B && C
- B && D

as opposed to only allowing, say, A && C and B && D. I don't think we have a way to specify such allowable strict configurations within a single ReferenceValues proto, but we could always try verifying using a list of ReferenceValues until the first one matches to achieve that kind of more strict checking. I don't think the commit really changes that situation much, given that we already have this issue with the other BinaryReferenceValues matchers.